notion2markdown/json2md.py

```diff
@@ -141,6 +149,20 @@ def apply_annotation(self, value, prv=None, nxt=None, annotation_to_mark={
                 state["annotations"] = []
             text = self.json2md(value[value["type"]])
             annotations = value.get("annotations", {})
+            empty = ""
+
+            def preserveTrailingSpace(line:str, mark:str):
+                tokens = re.split(r'(\s+)', line)
+                hasTrailingWhitespace = len(tokens) > 2 and re.match(r'\s+', tokens[-2]) and tokens[-1] == ""
+                if hasTrailingWhitespace:
+                    text = "".join(tokens[0:-2]).strip()
+                    return f'{text}{mark}{tokens[-2]}'
+                return f'{line}{mark}'
```

Comment:
First, thanks for the PR! Good catches. Instead of using `re`, we could actually use the rstrip'ed line.

```
def preserveTrailingSpace(line: str, mark: str):
    stripped = line.rstrip()  # line without any trailing whitespace
    whitespace = line[len(line.rstrip()):]  # only trailing whitespace (if any)
    return f"{stripped}{mark}{whitespace}"
```

[demo](https://pytwiddle.com?source=def+preserveTrailingSpace%28line%3A+str%2C+mark%3A+str%29%3A%0A++++stripped+%3D+line.rstrip%28%29++%23+line+without+any+trailing+whitespace%0A++++whitespace+%3D+line%5Blen%28line.rstrip%28%29%29%3A%5D++%23+only+trailing+whitespace+%28if+any%29%0A++++return+f%22%7Bstripped%7D%7Bmark%7D%7Bwhitespace%7D%22%0A++++%0A%0Aprint%28repr%28preserveTrailingSpace%28%27hello+world%27%2C+%27%5E%27%29%29%29%0Aprint%28repr%28preserveTrailingSpace%28%27hello+world+%27%2C+%27%5E%27%29%29%29) for the above snippet